packages/@react-spectrum/datepicker/test/DateField.test.js

```diff
@@ -218,4 +219,86 @@ describe('DateField', function () {
       }
     });
   });
+
+  describe('events', function () {
+    let onBlurSpy = jest.fn();
+    let onFocusChangeSpy = jest.fn();
+    let onFocusSpy = jest.fn();
+    let onKeyDownSpy = jest.fn();
+    let onKeyUpSpy = jest.fn();
+
+    afterEach(() => {
+      onBlurSpy.mockClear();
+      onFocusChangeSpy.mockClear();
+      onFocusSpy.mockClear();
+      onKeyDownSpy.mockClear();
+      onKeyUpSpy.mockClear();
+    });
+
+    it('should focus field and switching segments via tab does not change focus', function () {
+      let {getAllByRole} = render(<DateField label="Date" onBlur={onBlurSpy} onFocus={onFocusSpy} onFocusChange={onFocusChangeSpy} />);
+      let segments = getAllByRole('spinbutton');
+
+      expect(onBlurSpy).not.toHaveBeenCalled();
+      expect(onFocusChangeSpy).not.toHaveBeenCalled();
+      expect(onFocusSpy).not.toHaveBeenCalled();
+
+      userEvent.tab();
+      expect(segments[0]).toHaveFocus();
+
+      expect(onBlurSpy).not.toHaveBeenCalled();
+      expect(onFocusChangeSpy).toHaveBeenCalledTimes(1);
+      expect(onFocusSpy).toHaveBeenCalledTimes(1);
+
+      userEvent.tab();
+      expect(segments[1]).toHaveFocus();
+      expect(onBlurSpy).not.toHaveBeenCalled();
+      expect(onFocusChangeSpy).toHaveBeenCalledTimes(1);
+      expect(onFocusSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call blur when focus leaves', function () {
+      let {getAllByRole} = render(<DateField label="Date" onBlur={onBlurSpy} onFocus={onFocusSpy} onFocusChange={onFocusChangeSpy} />);
+      let segments = getAllByRole('spinbutton');
+
+      expect(onBlurSpy).not.toHaveBeenCalled();
+      expect(onFocusChangeSpy).not.toHaveBeenCalled();
+      expect(onFocusSpy).not.toHaveBeenCalled();
+
+      userEvent.tab();
+      expect(segments[0]).toHaveFocus();
+
+      userEvent.tab();
+      expect(segments[1]).toHaveFocus();
+
+      userEvent.tab();
+      expect(segments[2]).toHaveFocus();
+      expect(onBlurSpy).toHaveBeenCalledTimes(0);
+
+      userEvent.tab();
+      expect(onBlurSpy).toHaveBeenCalledTimes(1);
+      expect(onFocusChangeSpy).toHaveBeenCalledTimes(2);
+      expect(onFocusSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should trigger right arrow key event for segment navigation', function () {
+      let {getAllByRole} = render(<DateField label="Date" onKeyDown={onKeyDownSpy} onKeyUp={onKeyUpSpy} />);
+      let segments = getAllByRole('spinbutton');
+
+      expect(onKeyDownSpy).not.toHaveBeenCalled();
+      expect(onKeyUpSpy).not.toHaveBeenCalled();
+
+      userEvent.tab();
+      expect(segments[0]).toHaveFocus();
+      // mock tab via userEvent calls up, but not down
```

Comment:
This is incorrect, this is how the browser works https://codesandbox.io/s/frosty-hofstadter-w0v6q2?file=/src/App.js
the events are fired on the document.activeElement, which changes from keydown to keyup because focus moved on keydown